File: app/src/main/java/pl/expert/mobilewzr/data/Subject.java
package pl.expert.mobilewzr.data;

import com.univocity.parsers.annotations.Format;
import com.univocity.parsers.annotations.Parsed;

import java.util.Date;

public class Subject {

    @Parsed(index = 0)
    private String Title;

    @Format(formats = "dd/MM/yyyy")
    @Parsed(index = 1)
    private Date StartDate;

    @Parsed(index = 2)
    private String StartTime;

    @Format(formats = "dd/MM/yyyy")
    @Parsed(index = 3)
    private Date EndDate;

    @Parsed(index = 4)
    private String EndTime;

    @Parsed(index = 5)
    private String Description;

    @Parsed(index = 6)
    private String Location;

    public String getTitle() {
        return Title;
    }

    public void setTitle(String title) {
        Title = title;
    }

    public Date getStartDate() {
        return StartDate;
    }

    public void setStartDate(Date startDate) {
        StartDate = startDate;
    }

    public String getStartTime() {
        return StartTime;
    }

    public void setStartTime(String startTime) {
        StartTime = startTime;
    }

    public Date getEndDate() {
        return EndDate;
    }

    public void setEndDate(Date endDate) {
        EndDate = endDate;
    }

    public String getEndTime() {
        return EndTime;
    }

    public void setEndTime(String endTime) {
        EndTime = endTime;
    }

    public String getDescription() {
        return Description;
    }

    public void setDescription(String description) {
        Description = description;
    }

    public String getLocation() {
        return Location;
    }

    public void setLocation(String location) {
        Location = location;
    }
}

File: app/src/main/java/pl/expert/mobilewzr/data/model/Subject.kt
package pl.expert.mobilewzr.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.univocity.parsers.annotations.Format
import com.univocity.parsers.annotations.Parsed
import java.util.*

@Entity
data class Subject(
    @PrimaryKey var index: Int = 0,

    @Parsed(index = 0, defaultNullRead = "") var title: String = "",

    @Parsed(index = 1) @Format(formats = ["MM/dd/yyyy"]) var startDate: Date = Date(),

    @Parsed(index = 2, defaultNullRead = "") var startTime: String = "",

    @Parsed(index = 3) @Format(formats = ["MM/dd/yyyy"]) var endDate: Date = Date(),

    @Parsed(index = 4, defaultNullRead = "") var endTime: String = "",

    @Parsed(index = 5, defaultNullRead = "") var description: String = "",

    @Parsed(index = 6, defaultNullRead = "") var location: String = ""
)
